### ISSUE TITLE
autoincrement option does not work in mysql

### ISSUE THREAD

**Issue by edkim** [kunklejr/node-db-migrate#124]
I try 

	db.createTable("my_table", {
		id: {type: 'int', primaryKey: true, autoIncrement: true}
	}, callback);

and it doesn't set the auto_increment option in mysql

**Comment by wzrdtales** [comment]
Actually autoIncrement works together with mysql.
Reopen this if you this is still not fixed and you can specify an exact scenario where it does not work.

**Comment by Casen** [comment]
According to my current tests, this still does not work.

**Comment by Casen** [comment]
```
db.createTable("Users", {
    id: {type: 'int', primaryKey: true, autoIncrement: true}
}, callback);
```

results in

```
CREATE TABLE `Users` (
  `id` int(11) DEFAULT NULL,
)
```

No auto-increment specified. This results in Users being created without any IDs.

**Comment by wzrdtales** [comment]
do you really use the latest version of db-migrate?

please post output of

    db-migrate --version


Because this actually working, I have copy pasted your specified migration and this results in:
```[SQL] CREATE TABLE  `Users` (`id` INTEGER  PRIMARY KEY AUTO_INCREMENT)```

And one last question, which dbms you're using?

**Comment by wzrdtales** [comment]
@Casen 

Also the table within the dbms matches everything specified:
```
MariaDB [denis]> DESCRIBE Users
    -> ;
+-------+---------+------+-----+---------+----------------+
| Field | Type    | Null | Key | Default | Extra          |
+-------+---------+------+-----+---------+----------------+
| id    | int(11) | NO   | PRI | NULL    | auto_increment |
+-------+---------+------+-----+---------+----------------+
1 row in set (0.02 sec)
```

**Comment by Casen** [comment]
You know what, I made a mistake! I do not have primaryKey: true on that column. I am using the email as the primary key currently, but maybe I will just change that around.

Let me check this out and get back to you.

The version I am using is 0.9.9

**Comment by Casen** [comment]
Yes just confirmed, autoIncrement does nothing unless it is the primaryKey

**Comment by Casen** [comment]
I am using MYSQL btw

**Comment by wzrdtales** [comment]
Well then this is not exactly a failure of db-migrate. But in your migrations.

An auto incrementor in any database must be at least unique!
So either you define your column as primaryKey or unique key.

**Comment by Casen** [comment]
So if I defined my migration as follows:

```
db.createTable("Users", {
    id: {type: 'int', unique: true, autoIncrement: true}
}, callback);
```

It would correct auto increment the column without it being the primary key?

I obviously understand the column must be unique... I thought that was just a given with auto increment

**Comment by wzrdtales** [comment]
If auto incrementor would automatically adjust the keys this would lead to undefined behavior. This is clearly not wanted.

And no not yet, yet it needs to be the primaryKey. But this is not wanted like this, I'm going to change this.

**Comment by wzrdtales** [comment]
also see https://github.com/kunklejr/node-db-migrate/blob/master/lib/driver/mysql.js#L101

**Comment by wzrdtales** [comment]
@Casen 
The missing ability of adding auto incrementor to unique keys was applied and published as patch of unexpected behavior in v0.9.10
